executor/analyze.go

```diff
@@ -85,6 +86,11 @@ const (
 
 // Next implements the Executor Next interface.
 func (e *AnalyzeExec) Next(ctx context.Context, req *chunk.Chunk) error {
+	err0 := saveOptsForAutoAnalyze(e)
+	if err0 != nil {
+		logutil.Logger(ctx).Warn("Fail to save options for auto analyze")
```

Comment:
We can also output some warnings to the MySQL client. The session context held in `AnalyzeExec` can do this by calling `AnalyzeExec.ctx.GetSessionVars().StmtCtx.AppendWarning()`

Reply:
Good point. Done.